backend/lib/dbHelpers.js

```diff
@@ -27,8 +31,9 @@ module.exports = (client) => {
     })
     cb(null, productId)
   }
+  this.addProduct = Bluebird.promisify(addProduct)
 
-  this.getProductById = (id, cb) => {
+  getProductById = (id, cb) => {
     client.hgetall(id, (err, reply) => {
       if (err) {
         console.log(' --> was not able to retrieve info for product with id', id)
```

Comment:
After this line, you call `throw err`. This will crash the server. Is this desired behaviour? Maybe we can just reject the promise?